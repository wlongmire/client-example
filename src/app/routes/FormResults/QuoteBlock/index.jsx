import React, { Component, PropTypes } from 'react'
import ToggleDisplay from 'app/components/shared/ToggleDisplay'
import { commifyNumber } from 'app/utils/utilities'
import classNames from 'classnames'

export class QuoteBlock extends Component {
  constructor(props) {
    super(props)
    this.state = {}
  }

  render() {
    const {
      className, ratings, productTitle, mainTitle
    } = this.props
    console.log('THIS PROPS RATINGS xxxx', this.props.ratings)

    const regularTypeList = [
        { title: 'Total Premium', value: ratings.totalPremium },
        { title: 'Base Premium', value: ratings.premium },
        { title: 'Additional Coverage', value: ratings.additionalCoverage },
        { title: 'Terrorism Coverage', value: ratings.terrorPremium },
    ]

    const excessTypeList = [
          { title: 'Total Premium', value: ratings.totalExcessPremium },
          { title: 'Base Premium', value: ratings.excessPremium },
          { title: 'Terrorism Coverage', value: ratings.excessTerrorPremium }
    ]

    const quoteValues = (list) => {
      return list.map((item, idx) => {
        return (<ToggleDisplay
          key={idx}
          show={item.value && item.value > 0}
          render={() => (
            <div className="premiumNumber">
              {item.title}
              <span>${commifyNumber(item.value || 0)}</span>
            </div>)
          }
        />)
      })
    }

    return (
      <div>
        <div className={classNames('quoteBlock', className)}>
          {mainTitle !== null && <h4 className="quoteTitle">{mainTitle}</h4>}
          <h4>{productTitle}</h4>
          { quoteValues(regularTypeList) }
          {ratings.excessPremium > 0 &&
          <div className="">
            <h4>Excess</h4>
            { quoteValues(excessTypeList)}
          </div>}
        </div>
      </div>
    )
  }
}

QuoteBlock.propTypes = {
  className: PropTypes.string,
  ratings: PropTypes.object
}

export default QuoteBlock
