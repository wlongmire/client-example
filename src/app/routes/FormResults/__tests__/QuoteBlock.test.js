/* eslint-env node, jest */
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { QuoteBlock } from '../QuoteBlock'

describe('>>> QuoteBlock Component --- Snapshot', () => {
  it('+++ tests OI Quote with', () => {
    const renderedValue = shallow(
      <QuoteBlock
        additionalCoverage={125}
        className={'oi primaryPricing'}
        terrorismCoverage={225}
        title={'Owners Interest (standard rate)'}
        totalPremium={4850}
      />)
    expect(toJson(renderedValue)).toMatchSnapshot()
  })
})