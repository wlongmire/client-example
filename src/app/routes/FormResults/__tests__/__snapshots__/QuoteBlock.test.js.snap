// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>> QuoteBlock Component (OI with excess)--- Snapshot +++ tests OI Quote with 1`] = `
<div>
  <div
    className="quoteBlock oi primaryPricing"
  >
    <h4
      className="quoteTitle"
    >
      Discount, contingent on binding with Builder’s Risk Package
    </h4>
    <h4>
      Owners Interest
    </h4>
    <ToggleDisplay
      render={[Function]}
      show={true}
    />
    <ToggleDisplay
      render={[Function]}
      show={true}
    />
    <ToggleDisplay
      render={[Function]}
      show={true}
    />
    <ToggleDisplay
      render={[Function]}
      show={true}
    />
    <div
      className=""
    >
      <h4>
        Excess
      </h4>
      <ToggleDisplay
        render={[Function]}
        show={true}
      />
      <ToggleDisplay
        render={[Function]}
        show={true}
      />
      <ToggleDisplay
        render={[Function]}
        show={true}
      />
    </div>
  </div>
</div>
`;

exports[`>>> QuoteBlock Component (OI without excess)--- Snapshot +++ tests OI Quote with 1`] = `
<div>
  <div
    className="quoteBlock ocp primaryPricing"
  >
    <h4
      className="quoteTitle"
    >
      Discount, contingent on binding with Builder’s Risk Package
    </h4>
    <h4>
      Owners Contractors Protective
    </h4>
    <ToggleDisplay
      render={[Function]}
      show={true}
    />
    <ToggleDisplay
      render={[Function]}
      show={true}
    />
    <ToggleDisplay
      render={[Function]}
      show={true}
    />
    <ToggleDisplay
      render={[Function]}
      show={true}
    />
  </div>
</div>
`;
